pkg/crosscluster/logical/lww_row_processor.go

```diff
@@ -308,6 +308,12 @@ func (p *failureInjector) injectFailure() error {
 func (srp *sqlRowProcessor) HandleBatch(
 	ctx context.Context, batch []streampb.StreamEvent_KV,
 ) (batchStats, error) {
+	defer func() {
```

Comment:
why is this necessary?

Reply:
Oops. This was a debug code path. Panics in distsql processors are really hard to debug and the first pass of this branch panic'd when the txn was nil.